### DISCUSSION TITLE
SVG image is incorrect

### DISCUSSION THREAD

**Discussion by ashes1** [chillerlan/php-qrcode#156] [Q&A]
When I set outputType to SVG and svgViewBoxSize to some value, I get small qr-code (where module size=1px) inside big box (bix size equals svgViewBoxSize). What am I doing wrong?

![image](https://user-images.githubusercontent.com/1319202/200193478-041ebb10-4328-4a94-8b66-88c7c66355d2.png)

**Comment by codemasher** [comment]
if you want to generate the QR Code with a fixed size, try the `QROptions::$svgWidth` and `QROptions::$svgHeight` settings:

```php
$options = new QROptions([
	'svgWidth' => '370px',
	'svgHeight' => '370px',

	// ...
]);
```

(the [`viewBox`](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/viewBox) is an [entirely different beast](https://css-tricks.com/scale-svg/#aa-the-viewbox-attribute))

**Reply by ashes1** [reply]
There are no such options in QROptionsTrait. And they do not affect in my case. I am using version 4.3.4
`$options = new QROptions([
            'quietzoneSize' => 2,
            //'scale' => 1,
            'imageBase64' => false,
            'outputType' => QRCode::OUTPUT_MARKUP_SVG,
            'svgWidth' => '500px',
            'svgHeight' => '500px',
            'svgViewBoxSize' => 500
        ]);
        $qrCode = new QRCode($options);
        $res = $qrCode->render($data);`

**Reply by codemasher** [reply]
You're right, i should have mentioned that these are in the dev-main branch. However, you can extend the `QRMarkup` output class and adjust [the header](https://github.com/chillerlan/php-qrcode/blob/v4.3.x/src/Output/QRMarkup.php#L29-L30) to your liking. See [this example](https://github.com/chillerlan/php-qrcode/blob/c4bbb199b3b218f8691a6f748b3d5a05f0a950b8/examples/custom_output.php#L31-L36) on how to use custom output classes.
